### DISCUSSION TITLE
JIT mode doesn't compile negative margins when using bracket notation with specific values

### DISCUSSION THREAD

**Discussion by imjb87** [tailwindlabs/tailwindcss#4033] [Help] [answered]
I've recently updated to v2.1.1 and enabled JIT (which is awesome by the way), but I'm trying to generate a negative margin class on the fly such as `-mt-[100px]`, but it appears this is not being compiled into a class by TailwindCSS.

Unless I've missed it somewhere, there's not a lot of documentation around the JIT compiler yet, but if there's a way to achieve what I'm trying to do here without using mucky inline CSS, please help!

EDIT: Basically I'd like to know if the syntax is different when using bracket notation. I know how to generate a custom class using a global.css stylesheet, but this feels nasty too.

Thanks in advance.

**Comment by adamwathan** [comment, marked as answer]
Hey! Yeah for arbitrary values you just put the whole value in there:

`mt-[-100px]`

This felt like it made the most sense since you can do stuff like `mt-[calc(-100px+25vh)]` where positive and negative units would be mixed in there anyways.

**Reply by imjb87** [reply]
Oh what? That's awesome! So that means I can get rid of my nasty inline calcs too!

Thanks man!

**Reply by MattAppleton** [reply]
> Hey! Yeah for arbitrary values you just put the whole value in there:
> 
> `mt-[-100px]`
> 
> This felt like it made the most sense since you can do stuff like `mt-[calc(-100px+25vh)]` where positive and negative units would be mixed in there anyways.

So useful :)

**Reply by ghost** [reply]
good to know :)

**Comment by aerni** [comment]
I was just running into this today. I think it would still be nice to have a shortcut for simple negative values like `-mt-[100px]`. It feels more natural to the framework. It took me like an hour to figure out why this wasn't working.
